subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -161,4 +171,54 @@ public boolean isSatisfiedBy(Task element) {
 
         // TODO - check should depend on test
     }
+
+    private void configureTestedComponent(final Project project) {
+        project.getPlugins().withType(SwiftExecutablePlugin.class, new Action<SwiftExecutablePlugin>() {
+            @Override
+            public void execute(SwiftExecutablePlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+
+        project.getPlugins().withType(SwiftLibraryPlugin.class, new Action<SwiftLibraryPlugin>() {
+            @Override
+            public void execute(SwiftLibraryPlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+    }
+
+    private void configureTestedSwiftComponent(Project project) {
+        TaskContainer tasks = project.getTasks();
+        DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
+
+        SwiftCompile compile = tasks.create("compileMainSwiftWithTestingEnabled", SwiftCompile.class, configureCompileTestedComponentTask(tasks.withType(SwiftCompile.class).getByName("compileSwift")));
```

Comment:
Except if we have another plan for this, I may have to disagree with this decision. There are valid reasons why you would want to unit test a release variant such as the impact of the optimization on your code at runtime.

According to [Apple's documentation](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/testing_with_xcode/chapters/04-writing_tests.html#//apple_ref/doc/uid/TP40014132-CH4-SW11), it almost seems like an optional capability that can be added to the tested component's binaries. We can certainly build upon the variant to infrastructure and add the `enable testability` option to the `xctest` configuration DSL.

I will change it, but it's something to keep in mind as it will most likely be a blocker for some organization.